### PR TITLE
Group radsec clients when generating authorised clients

### PR DESCRIPTION
**Pull request by C-gyorfi**
image

### CODE REVIEW

File: app/lib/use_cases/generate_authorised_clients.rb [hunk 1]
```diff
@@ -1,15 +1,29 @@
 class UseCases::GenerateAuthorisedClients
   def call(clients:)
-    output = clients.map do |client|
+    output = clients.reject(&:radsec?).map do |client|
```
Comment (emileswarts) [inline app/lib/use_cases/generate_authorised_clients.rb:3]: think for performance reasons we could maybe push this into sql?
Operating on big lists in memory might have a performance impact.

def call(clients:, radsec_clients:)

and populating these in the controller would be:
clients = Client.where(radsec: false)
radsec_clients = Client.where(radsec: true)

File: app/lib/use_cases/generate_authorised_clients.rb [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
       client_definition(client)
     end
 
-    output.join("\n\n")
+    radsec_clients = clients.select(&:radsec?)
+
+    if radsec_clients.any?
+      output << "clients radsec {"
```
Comment (emileswarts) [inline app/lib/use_cases/generate_authorised_clients.rb:10]: bit of a mission to test but I wonder what would happen if a group exists without any clients in it. Then we could always print the group to the file regardless of whether there are radsec clients. Nitpicking sorry

File: app/lib/use_cases/generate_authorised_clients.rb (hunk 1 above)
Comment (C-gyorfi) [reply app/lib/use_cases/generate_authorised_clients.rb]: `radsec` is not a field in the db, but I made some changes in f65dd3e let me know what you think

File: app/lib/use_cases/generate_authorised_clients.rb (hunk 2 above)
Comment (C-gyorfi) [reply app/lib/use_cases/generate_authorised_clients.rb]: This was actually easier than you think 07b13db

**Review by emileswarts — APPROVED** [review]

**Pull request merged**